### ISSUE TITLE
gnutls make fails: missing separator in Makefile line 2042

### ISSUE THREAD

**Issue by LigH-de** [jb-alvarado/media-autobuild_suite#1058]
ab-suite.make.log:
```
CFLAGS: -mthreads -mtune=generic -O2 -pipe
LDFLAGS: -pipe -static-libgcc -static-libstdc++
make 
Makefile:2042: *** missing separator.  Stop.
Makefile:2042: *** missing separator.  Stop.
```

Makefile line 2042:
```
@CODE_COVERAGE_RULES@
```

You seem to clamp gnutls sources (#tag=gnutls_3_*) so I am not sure how I should report this to their issues; better you have a look first and suggest how to proceed.

**Comment by schmidthubert** [comment]
That's why I added IgnorePKg back to downgrade for the moment, here my comment:
https://github.com/jb-alvarado/media-autobuild_suite/pull/1036#issuecomment-455377300

The change is mentioned in the NEWS file, https://github.com/autoconf-archive/autoconf-archive,

* Noteworthy changes in release 2019.01.06 (2019-01-06) [stable]
...
  The macro AX_CODE_COVERAGE was modified to use AX_ADD_AM_MACRO_STATIC,
  and thus unfortunately usage was changed.

**Comment by Ekerim** [comment]
I started out with a completely clean master .ZIP as of today and got stuck on this same issue.
Is there anything a complete know-nothing like me can do to get past this issue?

[logs.zip](https://0x0.st/sCkK.zip)

**Comment by schmidthubert** [comment]
The curl team already [fixed it](https://github.com/curl/curl/commit/179311ec37ffed5b88f446f4348993a51a168abd) (but MABS is currently using LATEST tag for curl which will NOT have that change), harfbuzz team still sleeping like a baby and so is the gnutls team ... There you can also find the reason: "The previously used ax_code_coverage.m4 is not license compatible and must not be used." and "m4/ax_code_coverage.m4 conflicts with ISC license" 

The "previously" used ax_code_coverage.m4 has an md5sum of 82644b46e3f9c158b75914a2e86badbc so apparently everyone using that file needs to rework their stuff now ...

**Comment by Ekerim** [comment]
Thanks for explaining the problem to me, but again, is there anything I can do to work around this issue until it is fixed upstream ? Like exclude some program that pulls in the dependency?
If, for example, the reason I get this error is because I answered yes to compile curl... can I edit the files somehow to not do that and thus avoid the error ?
All I really need is ffmpeg & ffprobe with nvenc.

**Comment by 1480c1** [comment]
Goto your build folder, edit the media-autobuild_suite.ini and change curl=* to curl=2, it will disable building curl, however I think something else also builds gnutls.

**Comment by 1480c1** [comment]
It's also in ffmpeg options

**Comment by nille02** [comment]
gnutls is a requirement. i remove it from ffmpeg, no curl, no mpv etc. but it still get build. so until its fixed upstream no new builds :/

**Comment by Ekerim** [comment]
Thank you kuh3h3, I'll try that.
Yeah, sounds like a real ugly hack, it is there for a reason... but if it'll get me an ffmpeg I'll take it for now.

**Comment by Ekerim** [comment]
I noticed that as well @nille02, but as you said... not a problem.

**Comment by LigH-de** [comment]
@nille02 - ridiculous, yes: 12 min on my PC. I wonder what its purpose is, at all.

**Comment by nille02** [comment]
and @kuh3h3 describe a work around. if you do the changes while he is in the bootstrap step it runs without a problem.

**Comment by BlackScreen** [comment]
@nille02 The fix is not working for me.

```
09:38:16 ┌ gnutls git ............................... [Recently updated]
09:38:17 ├ Running bootstrap...
09:54:00 ├ Running configure...
10:05:11 ├ Running make...
Likely error:
CFLAGS: -mthreads -mtune=generic -O2 -pipe
LDFLAGS: -pipe -static-libgcc -static-libstdc++
make
Makefile:2042: *** missing separator.  Stop.
Makefile:2042: *** missing separator.  Stop.
make failed. Check E:/AB-Suite/build/gnutls-git/build-64bit/ab-suite.make.log
This is required for other packages, so this script will exit.
10:05:11   Creating diagnostics file...

All relevant logs have been anonymously uploaded to https://0x0.st/zrS0.zip
Copy and paste [logs.zip](https://0x0.st/zrS0.zip) in the GitHub issue.
Make sure the suite is up-to-date before reporting an issue. It might've been fixed already.
Try running the build again at a later time.
```

**Comment by LigH-de** [comment]
@BlackScreen - did you read the instructions carefully?

Did you manually edit the file **build\gnutls-git\Makefile.am** as soon as `Running bootstrap...` appeared in the console? Timing is the key here. This file is recreated each time you run the suite if gnutls unless it is compiled to the recent version.

**Comment by wiiaboo** [comment]
You're not supposed to manually edit files like that. The build files a. Create a patch and add it to apply before the build in the suite itself and if it works, PR it and it'll stay there.

**Comment by LigH-de** [comment]
Of course this kind of patchwork is not a solution ... I don't know about the others already involved here, but I lack both the time and the experience with e.g. sed and regular expressions (or even the knowledge about other more efficient solutions in the range of tools provided in MSYS2) to create one. Sorry, it's not my daily domain.

**Comment by kliffgomel** [comment]
12:39:56 ┌ gnutls git ............................... [Recently updated]
12:40:00 ├ Running bootstrap...
Likely error:
LDFLAGS: -pipe -static-libgcc -static-libstdc++
./bootstrap --skip-po
No gtk-doc support found. You can't build the docs.
error: pathspec 'tests/suite/tls-fuzzer/python-ecdsa' did not match any file(s) known to git
error: pathspec 'tests/suite/tls-fuzzer/tlsfuzzer' did not match any file(s) known to git
error: pathspec 'tests/suite/tls-fuzzer/tlslite-ng' did not match any file(s) known to git
./bootstrap: Bootstrapping from checked-out gnutls sources...
./bootstrap: consider installing git-merge-changelog from gnulib
./bootstrap: getting gnulib files...
error: pathspec 'gnulib' did not match any file(s) known to git
bootstrap failed.
https://0x0.st/zrkF.zip

Is this error also related to the topic being discussed?

**Comment by wiiaboo** [comment]
No, different issue, but also known: https://github.com/jb-alvarado/media-autobuild_suite/issues/1014

**Comment by Selur** [comment]
@pat357: can you give a line where to add
`sed -i.org 's:^@CODE_COVERAGE_RULES@::' Makefile.in`
?
Adding it to as line 292, before:
`log bootstrap ./bootstrap --skip-po`
doesn't fix the issue here.

**Comment by Selur** [comment]
Yeah, manually editing the Makefile.am while bootstrap is running seems to work, but changing the build script would be preferable to avoid having to monitor the script all the time,...

**Comment by schmidthubert** [comment]
https://gitlab.com/gnutls/gnutls/issues/691
